### PR TITLE
Add an abstraction layer for `AnnotatedServiceConfigSetters `

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

We recently had a consensus that we would like to provide a way to add context paths for a subset of services.
We will have to add another `AnnotatedServiceConfigSetters` for this purpose.
By adding an abstraction layer, we could dedup some of the logic for building annotated services.

Modifications:

- Introduce an `AbstractAnnotatedServiceConfigSetters` which extends setters.
- `AnnotatedServiceBindingBuilder` and `VirtualHostAnnotatedServiceBindingBuilder` now extend `AbstractAnnotatedServiceConfigSetters`.

Result:

- We can proceed with the next step of refactoring
- Final version: https://github.com/line/armeria/pull/5037

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/VirtualHostAnnotatedServiceBindingBuilder.java [hunk 1]
```diff
@@ -67,271 +54,211 @@
...
+            Function<? super HttpService, ? extends HttpService> decorator) {
+        return (VirtualHostAnnotatedServiceBindingBuilder) super.decorator(decorator);
     }
 
+    @SafeVarargs
     @Override
-    public VirtualHostAnnotatedServiceBindingBuilder maxRequestLength(long maxRequestLength) {
-        defaultServiceConfigSetters.maxRequestLength(maxRequestLength);
-        return this;
+    public final VirtualHostAnnotatedServiceBindingBuilder decorators(
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/VirtualHostAnnotatedServiceBindingBuilder.java:133]: nit
```suggestion
    public VirtualHostAnnotatedServiceBindingBuilder decorators(
```

**Review by ikhoon — APPROVED** [review]
Thanks, @jrhee17!

File: core/src/main/java/com/linecorp/armeria/server/VirtualHostAnnotatedServiceBindingBuilder.java (hunk 1 above)
Comment (jrhee17) [reply core/src/main/java/com/linecorp/armeria/server/VirtualHostAnnotatedServiceBindingBuilder.java:133]: This is actually necessary for `@SafeVarargs` 😄

**Review by minwoox — APPROVED** [review]
Nice abstraction. Thanks!

**Pull request merged**